util/helpers.py

```diff
@@ -1,8 +1,147 @@
+import sys
+
+def load_model(session, checkpoint_filename, drop_source_layers, use_cudnn):
+    r'''
+    if use_cudnn:
+        move all Adam vars to init_vars
+    if drop_souce_layers>0:
+        move all dropped layers to init_vars
+    '''
+    
+    # Load the checkpoint and put all variables into loading list
+    # we will exclude variables we do not wish to load and then
+    # we will initialize them instead
+    ckpt = tfv1.train.load_checkpoint(checkpoint_filename)
+    load_vars = set(tfv1.global_variables())
+    init_vars = set()
+    if use_cudnn:
+        # Identify the variables which we cannot load, and set them
+        # for initialization
+        for v in load_vars:
+            try:
+                ckpt.get_tensor(v.op.name)
+            except tf.errors.NotFoundError:
+                print("CUDNN variable not found:", v.op.name)
+                init_vars.add(v)
+
+        load_vars -= init_vars
+        
+        # Check that the only missing variables (i.e. those to be initialised)
+        # are the Adam moment tensors, if they aren't then we have an issue
+        if any('Adam' not in v.op.name for v in init_vars):
+            log_error('Tried to load a CuDNN RNN checkpoint but there were '
+                      'more missing variables than just the Adam moment '
+                      'tensors.')
+            sys.exit(1)
+
+    if drop_source_layers>0:
+        '''
+        The transfer learning approach here needs us to supply
+        the layers which we will to exclude from the source model.
+        Say we want to exclude all layers except for the first one,
+        we are dropping five layers total, so:
+        drop_source_layers=5
+        If we want to use all layers from the source model except the last one, we use this:
+        drop_source_layers=1
+        '''
+        if drop_source_layers>=6:
+            log_error('The model only has 6 layers, but you are trying to drop '
+                      'all of them or more than all of them. Continuing and '
+                      'dropping only 5 layers')
+        
+        dropped_layers = ['2', '3', 'lstm', '5', '6'][-1 * min(5, int(drop_source_layers)):]
+
+        # Initialize all variables needed for DS, but not loaded from ckpt
+        for v in load_vars:
+            if any(layer in v.op.name for layer in dropped_layers):
+                init_vars.add(v)
+                
+        load_vars -= init_vars
+
+    for v in init_vars:
+        print("Initializing variable from scratch:", v.op.name)
+    for v in load_vars:
+        print("Loading variable from model:", v.op.name)
+        
+    init_op = tfv1.variables_initializer(list(init_vars))
+    for v in list(load_vars):
+        v.load(ckpt.get_tensor(v.op.name), session=session)
+    session.run(init_op)
+
+
+def check_model(checkpoint_filename):
+    checkpoint = tf.train.get_checkpoint_state(FLAGS.checkpoint_dir, checkpoint_filename)
+    if not checkpoint:
+        return False
```

Comment:
Return `None` here instead of `False`. `None` signals absence of a value/object, but it's not obvious what `False` means here, and `True` is never returned.